csharp/src/Ice/Encoding.cs

```diff
@@ -2,14 +2,68 @@
 // Copyright (c) ZeroC, Inc. All rights reserved.
 //
 
+using System;
+
 namespace Ice
 {
+    /// <summary>The Ice encoding defines how Slice constructs are marshaled to and later unmarshaled from sequences
+    /// of bytes. An Encoding struct holds a version of the Ice encoding.</summary>
     [System.Serializable]
     public readonly struct Encoding : System.IEquatable<Encoding>
     {
+        // The encodings known to the Ice runtime. Don't reference them from other static variables!
+
+        /// <summary>Version 1.0 of the Ice encoding.</summary>
+        public static readonly Encoding V1_0 = new Encoding(1, 0);
+
+        /// <summary>Version 1.1 of the Ice encoding.</summary>
+        public static readonly Encoding V1_1 = new Encoding(1, 1);
+
+        /// <summary>Version 1.2 of the Ice encoding.</summary>
+        public static readonly Encoding V2_0 = new Encoding(2, 0);
+
+        /// <summary>The most recent version of the Ice encoding.</summary>
+        public static readonly Encoding Latest = new Encoding(1, 1); // for now, will upgrade to 2.0
+
+        /// <summary>The major version number of this version of the Ice encoding.</summary>
         public readonly byte Major;
+
+        /// <summary>The minor version number of this version of the Ice encoding.</summary>
         public readonly byte Minor;
 
+        /// <summary>Parses a string into an Encoding.</summary>
+        /// <param name="str">The string to parse.</param>
+        /// <returns>A new encoding.</returns>
+        public static Encoding Parse(string str)
+        {
+            (byte major, byte minor) = Util.ParseMajorMinorVersion(str, throwOnFailure: true)!.Value;
```

Comment:
I find this throwOnFailure parameter a bit strange, why not just always throw and catch the failure when we don't want the exception to propagate?

Reply:
Yes, will fix.

Reply:
Fixed - moved the parsing to Encoding.Parse.